badge_server/test_badges.py

```diff
@@ -334,6 +422,52 @@ def test_git_py3_fresh_nodeps(self):
         self.assertEqual(json_response['right_color'], '#44CC44')
         self.assertLinkUrl(package, json_response['whole_link'])
 
+    def test_pypi_mix_fresh_nodeps(self):
+        RECENT_SUCCESS_DATA_EXTRA = RECENT_SUCCESS_DATA + [
```

Comment:
I'm not sure if this data setup is interesting because you'd get the same result even if that continue statement wasn't there.

You probably need something like this (just an outline):
```python
APACHE_BEAM_GOOGLE_API_CORE_RECENT_FAILURE_3 = compatibility_store.CompatibilityResult(
    [
        package.Package('apache-beam[gcp]'),
        package.Package('google-api-core')
    ],
    python_major_version=3,  # apache-beam does not support Python 3
    status=compatibility_store.Status.FAILURE,
    timestamp=datetime.datetime.utcnow())

    def test_pypi_py2py3_fresh_no_deps_failure_in_unsupported_version():
       self.fake_store.save_compatibility_statuses(
            RECENT_SUCCESS_DATA + 
            [APACHE_BEAM_GOOGLE_API_CORE_RECENT_FAILURE_3])
       # rest of your test is fine, also need to do the same for github.

```

Reply:
It's expected to get a `SUCCESS` status if the continue statement gets hit; it's there to make sure we don't compare CompatibilityResults for versions that aren't supported, so I'm not sure if there is a way to make this test more interesting.

Reply:
I mean that this test will pass even without the continue statement, right? (you could try it). A test for a particular behavior  isn't very convincing if the code could not even be there and it still passes ;-)

My variation will fail without the continue statement.

Reply:
OH ok I misread your example. I get what you meant now.

Reply:
Done